cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/transformers/AbstractTransformerConversion.java

```diff
@@ -120,4 +122,81 @@ protected void addAliases(Identifiable<?> identifiable) {
             }
         }
     }
+
+    private static RegulatingControl getRegulatingControl(Context context, String regulatingControlId) {
+        Objects.requireNonNull(regulatingControlId);
+        return context.regulatingControlMapping().cachedRegulatingControls().get(regulatingControlId);
+    }
+
+    protected void addCgmesDataAsProperties(Identifiable<?> transformer, TapChanger rtc, TapChanger ptc, Context context) {
+        addRatioTapChangerProperties(transformer, rtc, context);
+        addPhaseTapChangerProperties(transformer, ptc, context);
+    }
+
+    private static void  addRatioTapChangerProperties(Identifiable<?> transformer, TapChanger rtc, Context context) {
+        if (rtc == null || rtc.getId() == null) {
+            return;
+        }
+
+        if (rtc.getRegulatingControlId() != null) {
+            String key = String.format("RatioTapChanger.%s.TapChangerControl", rtc.getId());
+            transformer.setProperty(key, rtc.getRegulatingControlId());
+        }
+
+        if (rtc.getHiddenCombinedTapChanger() != null) {
+            defineHiddenTapChangerProperties(transformer, rtc, rtc.getHiddenCombinedTapChanger(), "RatioTapChanger", context);
+        }
```

Comment:
Do you need this because it is done after combination? Can't it be done the same way for hidden and non hidden tap changers?

Reply:
they must be done in a different way for hidden and not-hidden. Hidden after combination do not exist in IIDM Network, and non-hidden properties are taken from Network object.